packages/core/admin/admin/src/content-manager/pages/ListView/components/BulkActionButtons/tests/index.test.js

```diff
@@ -155,3 +189,122 @@ describe('BulkActionsBar', () => {
+            },
+          },
+        });
+
+        return (
+          <QueryClientProvider client={client}>
+            <ThemeProvider theme={lightTheme}>
+              <IntlProvider locale="en" messages={{}} defaultLocale="en">
+                <Provider store={store}>
+                  <MemoryRouter>
+                    <Table.Root>{children}</Table.Root>
+                  </MemoryRouter>
+                </Provider>
+              </IntlProvider>
+            </ThemeProvider>
+          </QueryClientProvider>
+        );
+      },
+    }
+  ),
+});
+
+describe('ConfirmDialogPublishAll', () => {
+  beforeAll(() => {
+    server.listen();
+  });
+
+  afterAll(() => {
+    server.close();
+  });
+
+  it('should show a default message if there are not draft relations', async () => {
+    const { findByText, queryByText } = setupConfirmPublish();
+    expect(await findByText('Confirmation')).toBeInTheDocument();
+    expect(
+      queryByText('not published yet and might lead to unexpected behavior')
+    ).not.toBeInTheDocument();
+
+    expect(await findByText('Are you sure you want to publish these entries?')).toBeInTheDocument();
+  });
+
+  it('should show the warning message with just 1 draft relation and two entries', async () => {
+    server.use(
+      rest.get('*/countManyEntriesDraftRelations', (req, res, ctx) => {
+        return res.once(
+          ctx.status(200),
+          ctx.json({
+            data: 1,
+          })
+        );
+      })
+    );
+
+    const { findByText, getByTestId } = setupConfirmPublish();
+    expect(await findByText('Confirmation')).toBeInTheDocument();
+    const draftWarning = await getByTestId('draft-rel-warning');
+
+    expect(draftWarning.textContent).toBe('1  relation  out of 2  entries   is ');
+  });
+
+  it('should show the warning message with 2 draft relations and two entries', async () => {
+    server.use(
+      rest.get('*/countManyEntriesDraftRelations', (req, res, ctx) => {
+        return res.once(
+          ctx.status(200),
+          ctx.json({
+            data: 2,
+          })
+        );
+      })
+    );
+
+    const { findByText, getByTestId } = setupConfirmPublish();
+    expect(await findByText('Confirmation')).toBeInTheDocument();
+    const draftWarning = await getByTestId('draft-rel-warning');
+
+    expect(draftWarning.textContent).toBe('2  relations  out of 2  entries   are ');
+  });
+
+  it('should not show the Confirmation component if there is an error coming from the API', async () => {
+    const originalConsoleError = console.error;
+    console.error = jest.fn();
+    server.use(
+      rest.get('*/countManyEntriesDraftRelations', (req, res, ctx) => {
+        return res.once(
+          ctx.status(500),
+          ctx.json({
+            errorMessage: 'Error',
+          })
+        );
+      })
+    );
+
+    const { queryByText } = setupConfirmPublish();
+    await waitFor(() => expect(queryByText('Confirmation')).not.toBeInTheDocument());
```

Comment:
I would use `getByRole('dialog', { name: "Confirmation" })` instead, it might be worth giving this a read, it's a nice guide – https://testing-library.com/docs/queries/about/#priority

Reply:
Ok, your suggestion doesn't work but I saw, in the previous test for the `BulkActionsBar` that we use to check if the dialog is on the page, like that
`await waitFor(() => expect(screen.getByRole('dialog')).toBeInTheDocument());`